src/node_buffer.cc

```diff
@@ -18,13 +18,27 @@
     if (!(r)) return env->ThrowRangeError("out of range index");            \
   } while (0)
 
-#define ARGS_THIS(argT)                                                     \
-  Local<Object> obj = argT;                                                 \
-  size_t obj_length = obj->GetIndexedPropertiesExternalArrayDataLength();   \
-  char* obj_data = static_cast<char*>(                                      \
-    obj->GetIndexedPropertiesExternalArrayData());                          \
-  if (obj_length > 0)                                                       \
-    CHECK_NE(obj_data, nullptr);
+#define ARGS_THIS_DEC(name)                                                 \
+  size_t name##_length;                                                     \
+  char* name##_data;
+
+#define SPREAD_ARG(val, name)                                                 \
+  CHECK((val)->IsUint8Array());                                               \
+  Local<Uint8Array> name = (val).As<Uint8Array>();                            \
+  ArrayBuffer::Contents name##_c = name->Buffer()->GetContents();             \
+  size_t name##_offset = name->ByteOffset();                                  \
+  name##_length = name->ByteLength();                                         \
+  name##_data = static_cast<char*>(name##_c.Data()) + name##_offset;          \
+  if (name##_length > 0)                                                      \
+    CHECK_NE(name##_data, nullptr);
```

Comment:
Never mind, I think I misread it yesterday.  I was afraid about it doing the wrong thing in e.g. an if statement but you'd never use this macro there.